File: LevelSDKExampleApp/app/src/main/java/com/theshopatvsp/levelandroidsdk/ble/model/response/NukeRecordsPacket.java
package com.theshopatvsp.levelandroidsdk.ble.model.response;

/**
 * Created by andrco on 9/28/16.
 */
public class NukeRecordsPacket extends DataPacket {
    private int payload;

    public NukeRecordsPacket(int payload) {
        this.payload = payload;
    }

    public NukeRecordsPacket(byte[] packet) {
        if (packet != null && packet.length > 0) {
            payload = packet[0];
        }
    }

    public byte[] getPacket() {
        byte bytes[] = new byte[1];

        bytes[0] = 22;

        return bytes;
    }

    public int getPayload() {
        return payload;
    }
}

File: LevelIOSSdkExample/LevelIOSSdkExample/LevelIOSSdk/model/NukeRecordsPacket.swift
//
//  NukeRecordsPacket.swift
//  LevelIOSSdkExample
//
//  Created by Andrew Cook on 11/4/16.
//  Copyright © 2016 TheShop. All rights reserved.
//

import Foundation

class NukeRecordsPacket: DataPacket {
    var payload: Int
    
    override init() {
        self.payload = 0
        
        super.init()
    }
    
    convenience init(payload: Int) {
        self.init()
        
        self.payload = payload
    }
    
    convenience init(bytes: [UInt8]) {
        self.init()
        
        self.payload = Int(bytes[0])
    }
    
    override func getPacket() -> [UInt8] {
        var bytes = [UInt8]()
        
        bytes.append(UInt8(22))
        
        return bytes
    }
}
